Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/ADLSGen2Deploy.java

```diff
@@ -0,0 +1,252 @@
+                            this.cred.generateSharedKey(req, this.headerBuilder.getHeaderGroup(), this.createDirReqParams)
+                    ).getHeaderList())
+                    .doOnNext(
+                            resp -> {
+                                if (resp.getStatusLine().getStatusCode() != 201) {
+                                    Exceptions.propagate(new UnknownServiceException("Can not create directory to save artifact using ADLSGen2 storage type"));
+                                }
+                            }
+                    )
+                    .map(ignored -> new HttpPut(filePath))
+                    .flatMap(put -> http.request(put, null
+                            , this.createFileReqParams
+                            , this.headerBuilder.setAuthorization(
+                                    this.cred.generateSharedKey(put, this.headerBuilder.getHeaderGroup(), this.createFileReqParams)
+                            ).getHeaderList()))
+                    .doOnNext(
+                            resp -> {
+                                if (resp.getStatusLine().getStatusCode() != 201) {
+                                    Exceptions.propagate(new UnknownServiceException("Can not create artifact using adlsgen2 storage type"));
+                                }
+                            }
+                    )
+                    .map(ignore -> new HttpPatch(filePath))
+                    .flatMap(patch -> http.request(patch, entity
+                            , this.appendReqParams
+                            , this.headerBuilder
+                                    .setContentType("application/octet-stream")
+                                    .setContentLength(len)
+                                    .setAuthorization(this.cred.generateSharedKey(patch, this.headerBuilder.getHeaderGroup(), this.appendReqParams))
+                                    .getHeaderList())
+                    )
+                    .doOnNext(
+                            resp -> {
+                                if (resp.getStatusLine().getStatusCode() != 202) {
+                                    Exceptions.propagate(new UnknownServiceException("Can not append data to artifact using ADLSGen2 storage type"));
+                                }
+                            }
+                    )
+                    .map(ignore -> new HttpPatch(filePath))
+                    .flatMap(patch ->
+                            {
+                                List<NameValuePair> flushReqParams = this.flushReqParamsBuilder.setPosition(len).build();
+                                return http.request(patch, null
+                                        , flushReqParams
+                                        , this.headerBuilder
+                                                .setContentType("application/json")
+                                                .setContentLength("0")
```

Comment:
Why patching 0 here?

Reply:
not entity patched ,just flush operation

---
In reply to: [265846317](https://github.com/Microsoft/azure-tools-for-java/pull/2878#discussion_r265846317) [](ancestors = 265846317)